src/server/replica.cc

```diff
@@ -893,53 +884,6 @@ void Replica::RedisStreamAcksFb() {
   }
 }
 
-class AclCheckerClient : public ProtocolClient {
```

Comment:
This introduces a small bug if master is an older version that we cover in our tests. The problem is that if ACL of the `masteruser` changes  then `REPLCONF ACK` will start failing silently without noticing and replication will break :( 

I guess this setup is more probably since upgrading from one version to the other would involve a replica copying the data and then being promoted to master. An intermediate solution would be to keep this check for older version masters and deprecate/completely remove at some later point. 

I am open for suggestions

Reply:
this is fine, the use-case is not interesting.